spswarehouse/powerschool/powerschool_calpads.py

```diff
@@ -90,85 +98,42 @@ def download_calpads_report_for_school(self, school_full_name: str, submission_w
 
         if(submission_window == 'EOY'):
```

Comment:
This logic is a little gross. Consider something like the following that takes advantage of the fact that python functions are first class objects

```
report_mapping = {
  'EOY': {
    'SINC': self._download_eoy_report_for_student_incident_records_sinc,
    'SIRS': self.download_eoy_report_for...
  },
  'All Year': {
    'SENR": ...
  },
}

report_mapping[submission_window][calpads_report_appreciation](report_kwargs)
```

If you wanted to go a step further in cleaning this up, you could do:
```
report_mapping = {}
def register_report(submission_window, report, to_call, additional_params = {}):
   report_mapping[submission_window][report] = { "to_call": to_call, "params": {} }

register_report("EOY", "SINC", self._download_eoy...)
register_report("EOY", "SIRS", self._download_eoy...)
register_report("All Year", "SENR", self._download_eoy...)
```

Reply:
I ended up putting the dictionary in for the moment; it was fastest.  FYI @dunham-mike , I think I'm close to the final form on this one.